password/test/tasty/Validate.hs

```diff
@@ -29,18 +32,24 @@ testValidate =
   testGroup
     "validate"
     [ testProperty "Length of default character set is valid" $
-        (T.length defaultCharSet) === 95,
+        (length $ nub defaultCharacterSet) === 95,
+      testProperty "Length of special character set is valid" $
+        (length $ nub specialCharacterSet) === 33,
+      testProperty "defaultCharacterSet does not contain control characters" $
+        (all (not . isControl) defaultCharacterSet),
+      testProperty "All characters in defaultSet are all ord c < 127" $
+        (all (\c -> ord c < 127) defaultCharacterSet),
```

Comment:
Added test for `defaultCharacterSet` and `specialCharacterSet` which checks the following:

- It does not contain control characters
- Every character applied with `ord` is less than 127
- Cotains 95 unique characters